### PR TITLE
Merge dev soft restart 16feb

### PR DESCRIPTION
**Pull request by AdoAdoAdo**
bring latest changes from development to soft restart branch

### CODE REVIEW

File: epochStart/mock/economicsHandlerStub.go [hunk 1]
```diff
@@ -6,22 +6,34 @@ import (
 	"github.com/ElrondNetwork/elrond-go/process"
 )
 
+
```
Comment (iulianpascalau) [inline epochStart/mock/economicsHandlerStub.go:9]: Maybe remove L8 & L9?

File: factory/apiResolverFactory.go [hunk 2]
```diff
@@ -20,145 +20,276 @@ import (
...
+		TxCostHandler:      txCostHandler,
+		VmFactory:          vmFactory,
+		VmContainer:        vmContainer,
+		StakedValueHandler: totalStakedValueHandler,
+	}
+
+	return external.NewNodeApiResolver(apiArgs)
+}
+
+func createScQueryService(
```
Comment (iulianpascalau) [inline factory/apiResolverFactory.go:140]: Was this new code copied "by hand" after the merge? Not questioning the reason for this as this kind of merge is hard to do, just double checking.

**Review by iulianpascalau — DISMISSED** [review]

File: factory/apiResolverFactory.go (hunk 2 above)
Comment (AdoAdoAdo) [reply factory/apiResolverFactory.go:140]: part of it manually applied merge as it was previously in a different file, and part of it refactor

File: epochStart/mock/economicsHandlerStub.go (hunk 1 above)
Comment (AdoAdoAdo) [reply epochStart/mock/economicsHandlerStub.go:9]: done

File: factory/apiResolverFactory.go (hunk 2 above)
Comment (iulianpascalau) [reply factory/apiResolverFactory.go:140]: right

**Review by iulianpascalau — APPROVED** [review]

**Review by sasurobert — APPROVED** [review]

File: node/nodeRunner.go [hunk 3]
```diff
@@ -93,7 +94,7 @@ func (nr *nodeRunner) Start() error {
 		return err
 	}
 
-	logGoroutinesNumber(0)
+	//logGoroutinesNumber(0)
```
Comment (iulianpascalau) [inline node/nodeRunner.go]: commented code

File: node/nodeRunner.go [hunk 4]
```diff
@@ -1124,15 +1125,47 @@ func cleanupStorageIfNecessary(workingDir string, cleanupStorage bool) error {
...
 
+	err = copyDirectory(gasScheduleFolder, statsFolder)
+	log.LogIfError(err)
+
 	for _, configFile := range configs {
 		copySingleFile(statsFolder, configFile)
 	}
 }
 
+func copyDirectory(source string, destination string) error {
```
Comment (iulianpascalau) [inline node/nodeRunner.go:1140]: Some unit tests would be great. Can add a TODO here and others can write the tests as this PR goes into a feature branch

File: node/nodeRunner.go (hunk 3 above)
Comment (AdoAdoAdo) [reply node/nodeRunner.go]: done

File: node/nodeRunner.go (hunk 4 above)
Comment (AdoAdoAdo) [reply node/nodeRunner.go:1141]: added todo

**Review by iulianpascalau — DISMISSED** [review]

**Review by sasurobert — DISMISSED** [review]

**Review by stefanandy — DISMISSED** [review]
System Test Passed

**Review by iulianpascalau — DISMISSED** [review]

**Review by sasurobert — DISMISSED** [review]

**Review by sasurobert — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**